packages/mds-web-sockets/clients.ts

```diff
@@ -0,0 +1,49 @@
+import WebSocket from 'ws'
+import { WebSocketAuthorizer } from '@mds-core/mds-api-authorizer'
+import { AuthorizationError } from '@mds-core/mds-utils'
+import log from '@mds-core/mds-logger'
+
+export class Clients {
+  authenticatedClients: WebSocket[]
+
+  subList: { [key: string]: WebSocket[] }
+
+  public constructor() {
+    this.subList = { EVENTS: [], TELEMETRIES: [] }
+    this.authenticatedClients = []
+    this.saveClient = this.saveClient.bind(this)
+  }
+
+  public isAuthenticated(client: WebSocket) {
+    return this.authenticatedClients.includes(client)
+  }
+
+  public saveClient(entities: string[], client: WebSocket) {
+    if (!this.authenticatedClients.includes(client)) {
+      client.send('Not authenticated!')
+      return
+    }
+
+    const trimmedEntities = entities.map(entity => entity.trim())
+
+    trimmedEntities.forEach(async entity => {
+      try {
+        this.subList[entity].push(client)
+      } catch {
+        await log.error(`failed to push ${entity}`)
+      }
+    })
+  }
```

Comment:
`await` in `forEach` usually isn't what you want. Doesn't matter much here because only logging is async, but if someone else added other async actions here the behavior could get weird.

Reply:
I was thinking of that when I had to add that async modifier, I suppose I could do a Promise.all() with a map, it would just feel kinda odd because this shouldn't _usually_ be async...